File: frontend/src/api/maintenance.ts
/**
 * 数据库维护 API
 */

import request from '@/utils/request'
import type { ApiResponse } from '@/types/api'

/**
 * 备份请求
 */
export interface BackupRequest {
  connection_id: number
  database?: string
  tables?: string[]
  format?: 'sql' | 'csv'
}

/**
 * 备份响应
 */
export interface BackupResponse {
  backup_id: string
  download_url: string
  file_size: number
}

/**
 * 恢复请求
 */
export interface RestoreRequest {
  connection_id: number
  database?: string
  sql_file: string // SQL 文件内容
}

/**
 * 表维护操作
 */
export interface TableMaintenanceRequest {
  connection_id: number
  database: string
  table: string
  operation: 'optimize' | 'repair' | 'analyze'
}

/**
 * 备份数据库
 */
export async function backupDatabase(params: BackupRequest): Promise<BackupResponse> {
  const response = await request.post<ApiResponse<BackupResponse>>('/admin/backup', params)
  return response.data
}

/**
 * 恢复数据库
 */
export async function restoreDatabase(params: RestoreRequest): Promise<void> {
  await request.post<ApiResponse<void>>('/admin/restore', params)
}

/**
 * 表优化
 */
export async function optimizeTable(params: TableMaintenanceRequest): Promise<void> {
  await request.post<ApiResponse<void>>('/admin/tables/optimize', {
    database: params.database,
    table: params.table
  }, {
    params: { connection_id: params.connection_id }
  })
}

/**
 * 表修复
 */
export async function repairTable(params: TableMaintenanceRequest): Promise<void> {
  await request.post<ApiResponse<void>>('/admin/tables/repair', {
    database: params.database,
    table: params.table
  }, {
    params: { connection_id: params.connection_id }
  })
}

/**
 * 表分析
 */
export async function analyzeTable(params: TableMaintenanceRequest): Promise<void> {
  await request.post<ApiResponse<void>>('/admin/tables/analyze', {
    database: params.database,
    table: params.table
  }, {
    params: { connection_id: params.connection_id }
  })
}

File: backend/internal/models/maintenance.go
package models

// BackupRequest 备份请求
type BackupRequest struct {
	ConnectionID   int      `json:"connection_id" binding:"required"`
	Database       string   `json:"database"`
	Tables         []string `json:"tables,omitempty"`
	Format         string   `json:"format,omitempty"`          // sql, csv
	Compress       bool     `json:"compress"`                  // 是否 gzip 压缩
	SchemaOnly     bool     `json:"schema_only"`               // 仅结构
	MaxFileSizeMB  int      `json:"max_file_size_mb"`          // 单文件最大大小（MB），0 表示不拆分
}

// BackupResponse 备份响应
type BackupResponse struct {
	BackupID    string `json:"backup_id"`
	DownloadURL string `json:"download_url"`
	FileSize    int64  `json:"file_size"`
	TaskID      *int   `json:"task_id,omitempty"`
}

// RestoreRequest 恢复请求
type RestoreRequest struct {
	ConnectionID int    `json:"connection_id" binding:"required"`
	Database     string `json:"database,omitempty"`
	SQLFile      string `json:"sql_file" binding:"required"` // SQL 文件内容
}

// TableMaintenanceRequest 表维护操作请求
type TableMaintenanceRequest struct {
	ConnectionID int    `json:"connection_id" binding:"required"`
	Database     string `json:"database" binding:"required"`
	Table        string `json:"table" binding:"required"`
	Operation    string `json:"operation" binding:"required,oneof=optimize repair analyze"`
}
